### PR TITLE
Cleanup strain slice

### PR DESCRIPTION
**Pull request by at7**
First part of removing deprecated variation code. Remove StrainSlice related code where possible.

### Description
Remove deprecated code StrainSlice and StrainSliceAdaptor.
Update StrainSlice occurrences in core API to use Variation::StrainSlice
Add tests for remaining use of StrainSlice.

Follow up from this PR:
Deprecate AlignStrainSlice and IndividualSlice
Possibly move MappedSliceContainer and MappedSlice to Variation API?

### CODE REVIEW

File: modules/Bio/EnsEMBL/Slice.pm [hunk 1]
```diff
@@ -2461,33 +2460,6 @@ sub get_by_Individual{
 
 }
 
-=head2 get_by_strain
```
Comment (avullo) [inline modules/Bio/EnsEMBL/Slice.pm]: So this should have been removed quite well before.

**Comment by avullo** [comment]
So far, so good to me, much needed move towards decoupling the respective API responsibilities.
Coverage complain is not to be taken seriously in this case. I'm gonna merge after someone else has taken a look.

Thanks

File: modules/Bio/EnsEMBL/Slice.pm (hunk 1 above)
Comment (at7) [reply modules/Bio/EnsEMBL/Slice.pm]: Yes. Everything Individual related can go. I haven't found any dependencies in the web code. I will deprecate the Individual code in a separate PR.

Comment (avullo) [reply modules/Bio/EnsEMBL/Slice.pm]: That makes sense, thanks.

**Comment by avullo** [comment]
Thanks Anja,, waiting for the "Individual" PR too :+1:

File: modules/Bio/EnsEMBL/AlignStrainSlice.pm [hunk 2]
```diff
@@ -302,7 +302,7 @@ sub get_all_Slices {
     }
   my $indAdaptor = $dbVar->get_IndividualAdaptor();
   my $ref_name =  $indAdaptor->get_reference_strain_name;
-  my $ref_strain = Bio::EnsEMBL::StrainSlice->new(
+  my $ref_strain = Bio::EnsEMBL::Variation::StrainSlice->new(
```
Comment (avullo) [inline modules/Bio/EnsEMBL/AlignStrainSlice.pm]: Here we forgot to import the corresponding module I suspect

**Pull request merged**